lib/puppet/forge/repository.rb

```diff
@@ -109,7 +110,7 @@ def read_response(request, io = nil)
         end
 
         io.write(response.body) if io.respond_to? :write
-        response
+        return response
```

Comment:
This sounds like a bug in a test, as it's probably stubbing the `start` method, but not doing so faithfully. I wouldn't expect the source code to change though. Can we fix the test(s)?